### PR TITLE
ci: add more juicy github action runners

### PR DESCRIPTION
**Pull request by q9f**
## Summary of changes
uses build jet external runners. should not make any difference (except for speed).

## Reference issue to close (if applicable)
n/a

## Other information and links
n/a

## Change checklist
- [x] I have performed a self-review of my own code,
- [ ] I have made corresponding changes to the documentation,
- [ ] I have added tests that prove my fix is effective or that my feature works (if possible),
- [ ] I have made sure the [CHANGELOG](https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md) is up-to-date. All user-facing changes should be reflected in this document.

### CODE REVIEW

**Comment by LesnyRumcajs** [comment]
@q9f buildjets are billable whereas GH workers for public repositories are free. Unless a job is critical from duration point of view (e.g., takes 2h on regular GH workers) , there is no real point in doing so, no?

**Comment by hanabi1224** [comment]
One downside of buildjet to note is that, it's very slow in downloading the build cache, and timeout from time to time

**Comment by q9f** [comment]
It's an offer. If you are fine with GA, I will move on silently 😶‍🌫️

**Comment by hanabi1224** [comment]
I think the code coverage job can definitely benefit from buildjet, and we can even enable it for PR(s). We recently moved it to GH hosted runners and it takes ~50min for each run

**Comment by LesnyRumcajs** [comment]
> I think the code coverage job can definitely benefit from buildjet, and we can even enable it for PR(s). We recently moved it to GH hosted runners to save cost and it takes ~50min for each run

I don't think it is helpful to enable it for PRs. After a recent change, it's doing roughly the same stuff as the test step, and we are not tracking coverage changes. For a regular coverage step, running it for 50 minutes is fine because it's already on `main`, and it's not blocking anyone.

It would be helpful to use a juicy machine for https://github.com/ChainSafe/forest/blob/main/.github/workflows/docker.yml until https://github.com/ChainSafe/forest/issues/2476 is resolved. We could then easily support multi-platform docker images without many tricks. Then again, not many folks need a native image (for now), so it might not be worth it.

**Comment by q9f** [comment]
ok let me know which of the jobs will definitely not benefit from buildjet and I will revert them to GA

**Comment by LesnyRumcajs** [comment]
@q9f Looking at other runs, if we add only calibnet sync checks and lint checks then we will sit at ~10m per PR. That is to say, other jobs on GA can manage under this limit and the coverage is completely async so it's not an issue.

File: .github/workflows/autoupdate.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ on:
 jobs:
   autoupdate:
     name: autoupdate
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-latest
```
Comment (q9f) [inline .github/workflows/autoupdate.yml:10]: latest is now 22.04

https://github.blog/changelog/2022-11-09-github-actions-ubuntu-latest-workflows-will-use-ubuntu-22-04/

**Review by LesnyRumcajs — APPROVED** [review]
Thank you!

**Review by hanabi1224 — APPROVED** [review]

**Pull request merged**